lib/puppet/parser/functions/openldap_password.rb

```diff
@@ -1,14 +1,16 @@
 module Puppet::Parser::Functions
-  newfunction(:openldap_password, :type => :rvalue, :doc => <<-EOS
+  newfunction(
+    :openldap_password,
+    :type  => :rvalue,
+    :arity => -2,
```

Comment:
`:arity` is the parameter to specify number of arguments expected. Puppet then checks whether the passed parameters meet that expectation. It's not documented unfortunately, but it has docs in the code:

https://github.com/puppetlabs/puppet/blob/f069143cd086735fc88d22d5d76c2a279f0b020c/lib/puppet/parser/functions.rb#L127-L136